### ISSUE TITLE
Formatting issues 

### ISSUE THREAD

**Issue by JHag6694** [AdaCore/ada_language_server#1149]
Hello 

Here are 3 small formatting issues I am facing with ALS 23.0.18 (and older releases too)
Source code and project settings to reproduce the issue:  https://github.com/JHag6694/AdaTest

1 : Adding a pragma Obsolescent to 1 procedure triggers the package to be displayed as striked

2 :  Format selection on a single comment line deletes 2 trailing characters 

3 :  Adding a comment between begin and case make the case instruction to be right-shifted

Issue 1: 
![image](https://github.com/AdaCore/ada_language_server/assets/26746642/38567e7c-1ccd-496a-8db4-fdb92fb22906)

Issue 2: 
![Animation3](https://github.com/AdaCore/ada_language_server/assets/26746642/4a82c64a-220e-4929-b079-11980946d83c)

Issue3: 
![Animation3](https://github.com/AdaCore/ada_language_server/assets/26746642/5649f2e0-506d-46ab-8c7c-3d7046fd5b5e)

**Comment by reznikmm** [comment]
Thank you for the report. Good news is that issue 1 has been fixed already and will be in 23.0.19

**Comment by JHag6694** [comment]
VScode was updated with ALS 23.0.19 today, and issue #1 of my test case does not seem to be fixed.

**Comment by JHag6694** [comment]
VScode updated with 23.0.19 today, and issue #1 does not seem to be fixed.
Did I miss something !?

**Comment by reznikmm** [comment]
Indeed :( We will investigate this again. Meantime you can use this:
```ada
package Messages is

   procedure Hello_World
     with Obsolescent;

end Messages;
```

**Comment by reznikmm** [comment]
I'm closing this as it works with current development version. Should work in the next release.

**Comment by JHag6694** [comment]
Hello @reznikmm, 

FYI, according to my tests, issues #2 and #3 are not fixed in 24.0.1.
Did I miss something? 
Maybe you prefer a specific GitHub issue for each one.
Just let me know.
